<h2>Why is this an issue?</h2>
<p>Hard-coding a URI makes it difficult to test a program for a variety of reasons:</p>
<ul>
  <li> path literals are not always portable across operating systems </li>
  <li> a given absolute path may not exist in a specific test environment </li>
  <li> a specified Internet URL may not be available when executing the tests </li>
  <li> production environment filesystems usually differ from the development environment </li>
</ul>
<p>In addition, hard-coded URIs can contain sensitive information, like IP addresses, and they should not be stored in the code.</p>
<p>For all those reasons, a URI should never be hard coded. Instead, it should be replaced by a customizable parameter.</p>
<p>Further, even if the elements of a URI are obtained dynamically, portability can still be limited if the path delimiters are hard-coded.</p>
<p>This rule raises an issue when URIs or path delimiters are hard-coded.</p>
<h3>Exceptions</h3>
<p>This rule does not raise an issue when:</p>
<ul>
  <li> A constant path is relative and contains at most two parts. </li>
  <li> A constant path is used in an annotation </li>
  <li> A path is annotated </li>
</ul>
<h2>How to fix it</h2>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
public class Foo {
  public static final String FRIENDS_ENDPOINT = "/user/friends"; // Compliant path is relative and has only two parts

  public Collection&lt;User&gt; listUsers() {
    File userList = new File("/home/mylogin/Dev/users.txt"); // Noncompliant
    Collection&lt;User&gt; users = parse(userList);
    return users;
  }
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
public class Foo {
  // Configuration is a class that returns customizable properties: it can be mocked to be injected during tests.
  private Configuration config;
  public Foo(Configuration myConfig) {
    this.config = myConfig;
  }
  public Collection&lt;User&gt; listUsers() {
    // Find here the way to get the correct folder, in this case using the Configuration object
    String listingFolder = config.getProperty("myApplication.listingFolder");
    // and use this parameter instead of the hard coded path
    File userList = new File(listingFolder, "users.txt"); // Compliant
    Collection&lt;User&gt; users = parse(userList);
    return users;
  }
}
</pre>
<p>Exceptions examples:</p>
<pre>
public class Foo {
  public static final String FRIENDS_ENDPOINT = "/user/friends"; // Compliant path is relative and has only two parts

  public static final String ACCOUNT = "/account/group/list.html"; // Compliant path is used in an annotation

  @Value("${base.url}" + ACCOUNT)
  private String groupUrl;

  @MyAnnotation()
  String path = "/default/url/for/site"; // Compliant path is annotated

}
</pre>

