<h2>Why is this an issue?</h2>
<p>This rule allows you to track the usage of the <code>@SuppressWarnings</code> mechanism.</p>
<h3>Noncompliant code example</h3>
<p>With a parameter value of "unused" :</p>
<pre>
@SuppressWarnings("unused")
@SuppressWarnings("unchecked")  // Noncompliant
</pre>

