<h2>Why is this an issue?</h2>
<p>This rule verifies that single-line comments are not located at the ends of lines of code. The main idea behind this rule is that in order to be
really readable, trailing comments would have to be properly written and formatted (correct alignment, no interference with the visual structure of
the code, not too long to be visible) but most often, automatic code formatters would not handle this correctly: the code would end up less readable.
Comments are far better placed on the previous empty line of code, where they will always be visible and properly formatted.</p>
<h3>Noncompliant code example</h3>
<pre>
int a1 = b + c; // This is a trailing comment that can be very very long
</pre>
<h3>Compliant solution</h3>
<pre>
// This very long comment is better placed before the line of code
int a2 = b + c;
</pre>

