<h2>Why is this an issue?</h2>
<p>This rule allows banning certain classes.</p>
<h3>Noncompliant code example</h3>
<p>Given parameters:</p>
<ul>
  <li> className:java.lang.String </li>
</ul>
<pre>
String name;  // Noncompliant
</pre>

