<h2>Why is this an issue?</h2>
<p>Annotating interfaces or interface methods with <code>@Cache*</code> annotations is not recommended by the official Spring documentation. If you
use the weaving-based aspect (mode="aspectj"), the <code>@Cache*</code> annotations will be ignored, and no caching proxy will be created.</p>
<h3>What is the potential impact?</h3>
<ul>
  <li> <strong>Confusing Code</strong>: Developers may mistakenly believe that caching is in effect, leading to confusion and incorrect assumptions
  about application performance. </li>
</ul>
<p>This rule raises an issue when an interface or an interface method is annotated with a <code>@Cache*</code> annotation.</p>
<h2>How to fix it</h2>
<p>Move <code>@Cache*</code> annotation from interface or interface method to the concrete class.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
public interface ExampleService {

    @Cacheable("exampleCache") //non compliant, interface method is annotated with @Cacheable
    String getData(String id);
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
@Service
public class ExampleServiceImpl implements ExampleService {

    @Cacheable("exampleCache")
    @Override
    public String getData(String id) {
        // Implementation here
    }
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Spring - <a href="https://docs.spring.io/spring-framework/reference/integration/cache/annotations.html#cache-annotation-enable">Declarative
  Annotation-based Caching</a> </li>
</ul>

