<h2>Why is this an issue?</h2>
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all method names match a provided regular expression.</p>
<h3>Noncompliant code example</h3>
<p>With default provided regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
<pre>
public int DoSomething(){...}
</pre>
<h3>Compliant solution</h3>
<pre>
public int doSomething(){...}
</pre>
<h3>Exceptions</h3>
<p>Overriding methods are excluded.</p>
<pre>
@Override
public int Do_Something(){...}
</pre>

