<h2>Why is this an issue?</h2>
<p>Merging collapsible <code>if</code> statements increases the codeâ€™s readability.</p>
<h3>Noncompliant code example</h3>
<pre>
if (file != null) {
  if (file.isFile() || file.isDirectory()) {
    /* ... */
  }
}
</pre>
<h3>Compliant solution</h3>
<pre>
if (file != null &amp;&amp; isFileOrDirectory(file)) {
  /* ... */
}

private static boolean isFileOrDirectory(File file) {
  return file.isFile() || file.isDirectory();
}
</pre>

