<h2>Why is this an issue?</h2>
<p>A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many
things.</p>
<h3>Noncompliant code example</h3>
<p>With a maximum number of 4 parameters:</p>
<pre>
public void doSomething(int param1, int param2, int param3, String param4, long param5) {
...
}
</pre>
<h3>Compliant solution</h3>
<pre>
public void doSomething(int param1, int param2, int param3, String param4) {
...
}
</pre>
<h3>Exceptions</h3>
<p>Methods annotated with :</p>
<ul>
  <li> Spring’s <code>@RequestMapping</code> (and related shortcut annotations, like <code>@GetRequest</code>) </li>
  <li> JAX-RS API annotations (like <code>@javax.ws.rs.GET</code>) </li>
  <li> Bean constructor injection with <code>@org.springframework.beans.factory.annotation.Autowired</code> </li>
  <li> CDI constructor injection with <code>@javax.inject.Inject</code> </li>
  <li> <code>@com.fasterxml.jackson.annotation.JsonCreator</code> </li>
  <li> Micronaut’s annotations (like <code>@io.micronaut.http.annotation.Get</code>) </li>
</ul>
<p>may have a lot of parameters, encapsulation being possible. Such methods are therefore ignored.</p>
<p>Also, if a class annotated as a Spring component (like <code>@org.springframework.stereotype.Component</code>) has a single constructor, that
constructor will be considered <code>@Autowired</code> and ignored by the rule.</p>

