<h2>Why is this an issue?</h2>
<p>Appending <code>String.valueOf()</code> to a <code>String</code> decreases the code readability.</p>
<p>The argument passed to <code>String.valueOf()</code> should be directly appended instead.</p>
<h3>Noncompliant code example</h3>
<pre>
public void display(int i){
  System.out.println("Output is " + String.valueOf(i));    // Noncompliant
}
</pre>
<h3>Compliant solution</h3>
<pre>
public void display(int i){
  System.out.println("Output is " + i);                    // Compliant
}
</pre>

