<h2>Why is this an issue?</h2>
<p>There are several reasons for a method not to have a method body:</p>
<ul>
  <li> It is an unintentional omission, and should be fixed to prevent an unexpected behavior in production. </li>
  <li> It is not yet, or never will be, supported. In this case an <code>UnsupportedOperationException</code> should be thrown. </li>
  <li> The method is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override. </li>
</ul>
<h3>Noncompliant code example</h3>
<pre>
public void doSomething() {
}

public void doSomethingElse() {
}
</pre>
<h3>Compliant solution</h3>
<pre>
@Override
public void doSomething() {
  // Do nothing because of X and Y.
}

@Override
public void doSomethingElse() {
  throw new UnsupportedOperationException();
}
</pre>
<h3>Exceptions</h3>
<p>This does not raise an issue in the following cases:</p>
<ul>
  <li> Non-public default (no-argument) constructors </li>
  <li> Public default (no-argument) constructors when there are other constructors in the class </li>
  <li> Empty methods in abstract classes </li>
  <li> Methods annotated with <code>@org.aspectj.lang.annotation.Pointcut()</code> </li>
</ul>
<pre>
public abstract class Animal {
  void speak() {  // default implementation ignored
  }
}
</pre>

