<h2>Why is this an issue?</h2>
<p>Classes in the <code>sun.*</code> packages are considered implementation details, and are not part of the Java API.</p>
<p>They can cause problems when moving to new versions of Java because there is no backwards compatibility guarantee. Similarly, they can cause
problems when moving to a different Java vendor, such as OpenJDK.</p>
<p>Such classes are almost always wrapped by Java API classes that should be used instead.</p>
<h3>Noncompliant code example</h3>
<pre>
import sun.misc.BASE64Encoder; // Noncompliant
</pre>

