<h2>Why is this an issue?</h2>
<p>There are two possible syntaxes for a lambda having only one input parameter with an inferred type: with and without parentheses around that single
parameter. The simpler syntax, without parentheses, is more compact and readable than the one with parentheses, and is therefore preferred.</p>
<p><strong>Note</strong> that this rule is automatically disabled when the projectâ€™s <code>sonar.java.source</code> is lower than <code>8</code>.</p>
<h3>Noncompliant code example</h3>
<pre>
(x) -&gt; x * 2
</pre>
<h3>Compliant solution</h3>
<pre>
x -&gt; x * 2
</pre>

