<h2>Why is this an issue?</h2>
<p>While the ternary operator is pleasingly compact, its use can make code more difficult to read. It should therefore be avoided in favor of the more
verbose <code>if</code>/<code>else</code> structure.</p>
<h3>Noncompliant code example</h3>
<pre>
System.out.println(i&gt;10?"yes":"no");
</pre>
<h3>Compliant solution</h3>
<pre>
if (i &gt; 10) {
  System.out.println(("yes");
} else {
  System.out.println("no");
}
</pre>

