<p>Failing to explicitly declare the visibility of a member variable could result it in having a visibility you donâ€™t expect, and potentially leave it
open to unexpected modification by other classes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class Ball {
    String color="red";  // Noncompliant
}
enum A {
  B;
  int a;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
class Ball {
    private String color="red";  // Compliant
}
enum A {
  B;
  private int a;
}
</pre>
<h2>Exceptions</h2>
<p>Members annotated with <code>@VisibleForTesting</code> annotation are ignored, as it indicates that visibility has been purposely relaxed to make
the code testable.</p>
<pre>
class Cone {
  @VisibleForTesting
  Logger logger; // Compliant
}
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

