<h2>Why is this an issue?</h2>
<p>Because the <code>equals</code> method takes a generic <code>Object</code> as a parameter, any type of object may be passed to it. The method
should not assume it will only be used to test objects of its class type. It must instead check the parameterâ€™s type.</p>
<h3>Noncompliant code example</h3>
<pre>
public boolean equals(Object obj) { // Noncompliant
  MyClass mc = (MyClass)obj;
  // ...
}
</pre>
<h3>Compliant solution</h3>
<pre>
public boolean equals(Object obj) {
  if (obj == null)
    return false;

  if (this.getClass() != obj.getClass())
    return false;

  MyClass mc = (MyClass)obj;
  // ...
}
</pre>

