<h2>Why is this an issue?</h2>
<p>Rather than creating a boxed primitive from a <code>String</code> to extract the primitive value, use the relevant <code>parse</code> method
instead. It will be clearer and more efficient.</p>
<h3>Noncompliant code example</h3>
<pre>
String myNum = "12.2";

float f = (new Float(myNum)).floatValue();  // Noncompliant; creates &amp; discards a Float
</pre>
<h3>Compliant solution</h3>
<pre>
String myNum = "12.2";

float f = Float.parseFloat(myNum);
</pre>

