<h2>Why is this an issue?</h2>
<p>According to <a href="http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.9">the Java Language Specification-8.9</a>:</p>
<blockquote>
  <p>Nested enum types are implicitly <code>static</code>.</p>
</blockquote>
<p>So thereâ€™s no need to declare them <code>static</code> explicitly.</p>
<h3>Noncompliant code example</h3>
<pre>
public class Flower {
  static enum Color { // Noncompliant; static is redundant here
    RED, YELLOW, BLUE, ORANGE
  }

  // ...
}
</pre>
<h3>Compliant solution</h3>
<pre>
public class Flower {
  enum Color { // Compliant
    RED, YELLOW, BLUE, ORANGE
  }

  // ...
}
</pre>

