<p>The use of escape sequences is mostly unnecessary in text blocks.</p>
<h2>Noncompliant Code Example</h2>
<p><code>\n</code> can be replaced by simply introducing the newline, <code>\"\"\"</code> it is sufficient to escape only the first qoute.</p>
<pre>
String textBlock = """
        \"\"\" this \nis
        text  block!
        !!!!
      """;
</pre>
<h2>Compliant Solution</h2>
<pre>
String textBlock = """
        \""" this
        is
        text  block!
        !!!!
      """;
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://openjdk.java.net/jeps/378">JEP 378: Text Blocks</a> </li>
  <li> <a href="https://cr.openjdk.java.net/~jlaskey/Strings/TextBlocksGuide_v9.html">Programmerâ€™s Guide To Text Blocks</a>, by Jim Laskey and Stuart
  Marks </li>
</ul>

