<p>If a string fits on a single line, without concatenation and escaped newlines, you should probably continue to use a string literal.</p>
<h2>Noncompliant Code Example</h2>
<pre>
String question = """
              What's the point, really?""";
</pre>
<h2>Compliant Solution</h2>
<pre>
String question = "What's the point, really?";
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://openjdk.java.net/jeps/378">JEP 378: Text Blocks</a> </li>
  <li> <a href="https://cr.openjdk.java.net/~jlaskey/Strings/TextBlocksGuide_v9.html">Programmerâ€™s Guide To Text Blocks</a>, by Jim Laskey and Stuart
  Marks </li>
</ul>

