<p>When testing exception via <code>@Test</code> annotation, any code in the test raising an exception of the same expected type will make it pass,
without making explicit what is exactly tested.</p>
<p> In addition, any code after the raised exception will not be executed. It will prevent you to test the state of the program after the raised
exception and, at worst, make you&nbsp;misleadingly think that it is executed.</p>
<p>You should consider using <a
href="https://github.com/junit-team/junit4/wiki/Exception-testing#using-assertthrows-method">org.junit.Assert.assertThrows</a> instead.</p>
<p> Alternatively, you could use <a href="https://github.com/junit-team/junit4/wiki/Exception-testing#trycatch-idiom">try-catch idiom</a> for JUnit
version &lt; 4.13 or if your project does not support lambdas.</p>
<h2>Noncompliant Code Example</h2>
<pre>
@Test(expected = IndexOutOfBoundsException.class)
public void testToString() {
  get().toString(); // Do you expect get() or toString() throwing the exception?
}

@Test(expected = IndexOutOfBoundsException.class)
public void testShouldFail() {
  get();
  // This test pass since execution will never get past this line.
  Assert.assertEquals(0, 1);
}

private Object get() {
  throw new IndexOutOfBoundsException();
}
</pre>
<h2>Compliant Solution</h2>
<p> * For JUnit &gt;= 4.13, use <a
href="https://github.com/junit-team/junit4/wiki/Exception-testing#using-assertthrows-method">org.junit.Assert.assertThrows</a>:</p>
<pre>
// This test correctly fails.
@Test
public void testToString() {
    Object obj = get();
    Assert.assertThrows(IndexOutOfBoundsException.class, () -&gt; obj.toString());
} {code}

 * For JUnit &lt; 4.13, use the [try-catch idiom|https://github.com/junit-team/junit4/wiki/Exception-testing#trycatch-idiom]:
{code:java}
<p>@Test</p>
<p>public void testShouldFail() {</p>
<p>    Object obj = get();</p>
<p>    try {</p>
<p>        obj.toString();</p>
<p>        Assert.fail("Expected an IndexOutOfBoundsException to be thrown");</p>
<p>    } catch (IndexOutOfBoundsException e) {}</p>
<p>    Assert.assertEquals(0, 1); // Correctly fails.</p>
<p>}</p>
<pre>
</pre>

<h2>See</h2>
<p> * <a href="https://github.com/junit-team/junit4/wiki/Exception-testing">JUnit exception testing documentation</a></p>
</pre>

