<p>Testing equality or nullness inside <code>org.junit.Assert.assertTrue()</code> should be simplified to its dedicated assertion.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Assert.assertTrue(a.equals(b));
Assert.assertTrue(a == b);
Assert.assertTrue(a == null);
Assert.assertTrue(a != null);
</pre>
<h2>Compliant Solution</h2>
<pre>
Assert.assertEquals(a, b);
Assert.assertSame(a, b);
Assert.assertNull(a);
Assert.assertNotNull(a);
</pre>

