<p>If an alternative in a regular expression only matches things that are already matched by another alternative, that alternative is redundant and
serves no purpose.</p>
<p>In the best case this means that the offending subpattern is merely redundant and should be removed. In the worst case itâ€™s a sign that this regex
does not match what it was intended to match and should be reworked.</p>
<h2>Noncompliant Code Example</h2>
<pre>
"[ab]|a"   // The "|a" is redundant because "[ab]" already matches "a"
".*|a"     // .* matches everything, so any other alternative is redundant
</pre>
<h2>Compliant Solution</h2>
<pre>
"[ab]"
".*"
</pre>

