<p>AssertJ assertions methods targeting the same object can be chained instead of using multiple <code>assertThat</code>. It avoids duplication and
increases the clarity of the code.</p>
<p>This rule raises an issue when multiples <code>assertThat</code> target the same tested value.</p>
<h2>Noncompliant Code Example</h2>
<pre>
assertThat(someList).hasSize(3);
assertThat(someList).contains("something");
</pre>
<h2>Compliant Solution</h2>
<pre>
assertThat(someList)
  .hasSize(3)
  .contains("something");
</pre>

