<h2>Why is this an issue?</h2>
<p><code>@Autowired</code> is an annotation in the Spring Framework for automatic dependency injection. It tells Spring to automatically provide the
required dependencies (such as other beans or components) to a class’s fields, methods, or constructors, allowing for easier and more flexible
management of dependencies in a Spring application. In other words, it’s a way to wire up and inject dependencies into Spring components
automatically, reducing the need for manual configuration and enhancing modularity and maintainability.</p>
<p>When using the Spring framework for dependency injection, it’s essential to use the <code>@Autowired</code> annotation judiciously. Specifically,
avoid using the <code>@Autowired</code> annotation on multiple constructors within a Spring component (e.g., a class annotated with
<code>@Component</code>). The <code>@Autowired</code> annotation indicates which constructor Spring should use for dependency injection. When multiple
constructors have this annotation, Spring may not be able to determine which one to use, leading to ambiguity in the bean creation process.</p>
<h2>How to fix it</h2>
<p>To maintain code clarity and ensure that the Spring context can create beans correctly, have only one constructor annotated with
<code>@Autowired</code> within a Spring component.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
@Component
public class MyComponent {
    private final MyService myService;

    @Autowired
    public MyComponent(MyService myService) {
        this.myService = myService;
    }

     @Autowired                                           // Noncompliant
    public MyComponent(AnotherService anotherService) {
        // ...
    }
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
@Component
public class MyComponent {
    private final MyService myService;

    @Autowired
    public MyComponent(MyService myService) {
        this.myService = myService;
    }

    public MyComponent(AnotherService anotherService) {
        // ...
    }
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> <a href="https://docs.spring.io/spring-framework/reference/core/beans/annotation-config/autowired.html">Spring Framework - Using @Autowired</a>
  </li>
</ul>
<h3>Articles &amp; blog posts</h3>
<ul>
  <li> <a href="https://www.baeldung.com/spring-autowire">Baeldung - Guide to Spring @Autowired</a> </li>
</ul>

