<p>Using <code>String.format()</code>, a method call or concatenation as a log message has a negative impact on the performance.</p>
<p>Change the log message into a String with {}-placeholders.</p>

<h2>Noncompliant Code Example</h2>
<pre>
logger.warn("Do not concatenate with " + someString);
logger.error(String.format("Do not format with %s", someVar));
logger.error(doNotGetLogMessageByMethodcall());
logger.error("Do not concatenate with " + someObject);
</pre>

<h2>Compliant Solution</h2>
<pre>
logger.debug("This is how you should use logging with {}", someVar);
</pre>

<h2>See</h2>
<ul>
	<li><a href="http://www.slf4j.org/faq.html#logging_performance">http://www.slf4j.org/faq.html#logging_performance</a></li>
</ul>
