<p>This Object.finalize() method is called by the garbage collector on an object when garbage collection determines that there are no more references to the object. But there is absolutely no warranty that this method will be called AS SOON AS the last references to the object are removed. It can be few microseconds to few minutes later. So when some system resources need to be disposed by an object, it's better to not rely on this asynchronous mechanism to dispose them.</p>
<h2>Noncompliant Code Example</h2>

<pre>
public class MyClass {
  ...
  protected void finalize() {
    releaseSomeResources();    //Non-Compliant
  }
  ...
}
</pre>
<h2>See</h2>

<ul>
<li> CERT Java Secure Coding, MET12-J - Do not use finalizers|</li>
</ul>

