<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all method names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>

<p>With default provided regular expression: XXXXX</p>
<pre>
public int DoSomething(){...}
</pre>
<h2>Compliant Solution</h2>

<pre>
public int doSomething(){...}
</pre>
<h2>Exceptions</h2>

<p>Overriding methods are excluded. </p>
<pre> 
@Override 
public int Do_Something(){...} 
</pre>

