<p>A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many things.</p>
<h2>Noncompliant Code Example</h2>

<p>The following code snippet illustrates this rule with a maximum number of 4 parameters:</p>
<pre>
public void doSomething(int param1, int param2, int param3, String param4, long param5) {
...
}
</pre>
<h2>Compliant Solution</h2>

<pre>
public void doSomething(int param1, int param2, int param3, String param4) {
...
}
</pre>

