<p>If a label is declared but not used in the program, it can be considered as dead code and should therefore be removed.</p>
<p>This will improve maintainability as developers will not wonder what this label is used for.</p>
<h2>Noncompliant Code Example</h2>

<pre>
&amp;lt;&amp;lt;foo&amp;gt;&amp;gt; -- Non-Compliant
BEGIN
  DBMS_OUTPUT.PUT_LINE('Hello, world!');
END;
</pre>
<h2>Compliant Solution</h2>

<pre>
-- Compliant
BEGIN
  DBMS_OUTPUT.PUT_LINE('Hello, world!');
END;
</pre>
<h2>See</h2>

<ul>
<li> MISRA C:2012, 2.6 - A function should not contain unused label declarations</li>
</ul>

