<p>The long suffix should always be written in uppercase, i.e. 'L', as the lowercase 'l' can easily be confused with the digit one '1'.</p>

<h2>Noncompliant Code Example</h2>
<pre>
long n = 10l;  // Noncompliant; easily confused with one zero one
</pre>

<h2>Compliant Solution</h2>
<pre>
long n = 10L; 
</pre>

<h2>See Also</h2>
<ul>
<li> {rule:squid:S818} - Literal suffixes should be upper case
</li></ul>

<h2>Deprecated</h2>
<p>This rule is deprecated, use {rule:squid:S818} instead.</p>

