<p><code>TODO</code> tags are commonly used to mark places where some more code is required, but which the developer wants to implement later.</p>
<p>Sometimes the developer will not have the time or will simply forget to get back to that tag.</p>
<p>This rule is meant to track those tags, and ensure that they do not go unnoticed.</p>
<h2>Noncompliant Code Example</h2>

<pre>
void doSomething() {
  // TODO
}
</pre>

