<p>Having too many return statements in a method increases the method's essential complexity because the flow of execution is broken each time a return statement is encountered. This makes it harder to read and understand the logic of the [method|function].</p>
<h2>Noncompliant Code Example</h2>

<p>The following code snippet illustrates this rule with the default threshold of 3:</p>
<pre>
public boolean myMethod() { // Non-Compliant as there are 4 return statements
  if (condition1) {
    return true;
  } else {
    if (condition2) {
      return false;
    } else {
      return true;
    }
  }
  return false;
}
</pre>

