<p>Overriding a method just to call the same method from the super class without performing any other actions is useless and misleading.</p>
<h2>Noncompliant Code Example</h2>

<pre>
public void doSomething() {                
  super.doSomething();
}

@Override
public boolean isLegal(Action action) {      
  return super.isLegal(action);
}
</pre>
<h2>Compliant Solution</h2>

<pre>
@Override
public boolean isLegal(Action action) {         // Compliant - not simply forwarding the call
  return super.isLegal(new Action(/* ... */));
}

@Id
@Override
public int getId() {                            // Compliant - there is annotation different from @Override
  return super.getId();
}
</pre>

