<p>Through Java's evolution keywords have been added. While code that uses those words as identifiers may be compilable under older versions of Java,
it will not be under modern versions. </p>
<table>
  <tbody>
    <tr>
      <th>Keyword</th>
      <th>Added</th>
    </tr>
    <tr>
      <td>assert</td>
      <td>1.4</td>
    </tr>
    <tr>
      <td>const</td>
      <td>reserved</td>
    </tr>
    <tr>
      <td>enum</td>
      <td>5.0</td>
    </tr>
    <tr>
      <td>goto</td>
      <td>reserved</td>
    </tr>
    <tr>
      <td>strictfp</td>
      <td>1.2</td>
    </tr>
  </tbody>
</table>
<h2>Noncompliant Code Example</h2>
<pre>
public void doSomething() {
  int enum = 42;            // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public void doSomething() {
  int magic = 42;
}
</pre>

