<p><code>java.lang.Error</code> and its subclasses represent abnormal conditions, such as <code>OutOfMemoryError</code>, which should only be encountered by the Java Virtual Machine.</p>
<h2>Noncompliant Code Example</h2>

<pre>
public class MyException extends Error { /* ... */ }       // Non-Compliant
</pre>
<h2>Compliant Solution</h2>

<pre>
public class MyException extends Exception { /* ... */ }   // Compliant
</pre>

