<p>
  Non-abstract <code>class</code>es and <code>enum</code>s with non-<code>static</code>,
  <code>private</code> members should explicitly initialize those members,
  either in a constructor or with a default value.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
class A { // Noncompliant
  private int field;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
class A {
  private int field;

  A(int field) {
    this.field = field;
  }
}
</pre>
