<p>This rule allows you to track the usage of the <code>@SuppressWarnings</code> mechanism. You can use one of two modes:</p>
<ol>
<li>
<code>BLACKLIST</code> - <code>listOfWarnings</code> stores not allowed suppressions, this is a default mode
<h2>Noncompliant Code Example</h2>
<p>In case &quot;listOfWarnings=unused&quot;</p>
<pre>
@SuppressWarnings("unused")
</pre>
<h2>BLACKLIST Compliant Solution</h2>
<p>In case &quot;listOfWarnings=all&quot;</p>
<pre>
@SuppressWarnings("unchecked")
@SuppressWarnings("unused")
</pre>
</li>
<li>
<code>WHITELIST</code> - <code>listOfWarnings</code> stores allowed suppressions
<h2>Noncompliant Code Example</h2>
<p>In case &quot;listOfWarnings=serial&quot;</p>
<pre>
@SuppressWarnings("unused")
</pre>
<h2>Compliant Solution</h2>
<p>In case &quot;listOfWarnings=serial&quot;</p>
<pre>
@SuppressWarnings("serial")
</pre>
</li>
</ol>
