<p>This rule allows you to track the usage of the <code>@SuppressWarnings</code> mechanism.</p>
<h2>Noncompliant Code Example</h2>

<p>In case "listOfWarnings=unused" :</p>
<pre>
@SuppressWarnings("unused")
</pre>
<h2>Compliant Solution</h2>

<p>In case "listOfWarnings=all" (default value) :</p>
<pre>
@SuppressWarnings("unchecked")
@SuppressWarnings("unused")
</pre>

