<p>This rule allows you to track the usage of the <code>@SuppressWarnings</code> mechanism.</p>

<h2>Noncompliant Code Example</h2>
<p>With a parameter value of "unused" :</p>
<pre>
@SuppressWarnings("unused")
</pre>

<h2>Compliant Solution</h2>
<p>With the default parameter value of "all":</p>
<pre>
@SuppressWarnings("unchecked")
@SuppressWarnings("unused")
</pre>

