<p>This rule allows you to track the usage of the <code>@SuppressWarnings</code> mechanism.</p>
<h2>Noncompliant Code Example</h2>
<p>In case &quot;listOfWarnings=unchecked&quot;</p>
<pre>
@SuppressWarnings("all")
</pre>
<h2>Compliant Solution</h2>
<p>In case &quot;listOfWarnings=unchecked&quot;</p>
<pre>
@SuppressWarnings("unchecked")
</pre>
