<p>
Integer literals starting with a zero are octal rather than decimal values.
While using octal values in Java is fully supported, most developers do not have experience with them.
They may not recognize octal values as such, mistaking them instead for decimal values.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
int myNumber = 010;   // myNumber will hold 8, not 10 - was this really expected?
</pre>

<h2>Compliant Solution</h2>
<pre>
int myNumber = 8;
</pre>
