<p>When the value of a private field is always assigned to in a class' methods before being read, then it is not being used to store class
information. Therefore, it should become a local variable in the relevant methods to prevent any misunderstanding.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class Foo {
  private int singularField;

  public void doSomething(int y) {
    singularField = y + 5;
    ...
    if(singularField == 0 {...}
    ...
  }

  public void doSomethingElse(int y) {
    singularField = y + 3;
    ...
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class Foo {

  public void doSomething(int y) {
    int singularField = y + 5;
    ...
  }

  public void doSomethingElse(int y) {
    int singularField = y + 3;
    ...
  }
}
</pre>
<h2>Exceptions</h2>
<p>This rule doesn't raise any issue on annotated field.</p>

