<p>If a local variable is declared but not used, it is dead code and should be removed. Doing so will improve maintainability because developers will not wonder what the variable is used for.</p>
<h2>Noncompliant Code Example</h2>

<pre>
public int numberOfMinutes(int hours) {
  int seconds = 0;   // seconds is never used 
  return hours * 60;
}
</pre>
<h2>Compliant Solution</h2>

<pre>
public int numberOfMinutes(int hours) {
  return hours * 60;
}
</pre>

