<p>Method/constructor references are more compact and readable than using lambdas, and are therefore preferred. Similarly, <code>null</code> checks
can be replaced with references to the <code>Objects::isNull</code> and <code>Objects::nonNull</code> methods.</p>
<p><strong>Note</strong> that this rule is automatically disabled when the project's <code>sonar.java.source</code> is lower than <code>8</code>.</p>
<h2>Noncompliant Code Example</h2>
<pre>
  List&lt;Integer&gt; list = new ArrayList&lt;Integer&gt;();
  list.add(0);
  list.add(1);
  list.add(2);

  list.forEach(n -&gt; { System.out.println(n); });
</pre>
<h2>Compliant Solution</h2>
<pre>
  List&lt;Integer&gt; list = new ArrayList&lt;Integer&gt;();
  list.add(0);
  list.add(1);
  list.add(2);

  list.forEach(System.out::println);
</pre>

