<p>Invoking a method designed to return a string representation of an object which is already a string is a waste of keystrokes. This redundant construction may be optimized by the compiler, but will be confusing in the meantime.</p>

<h2>Noncompliant Code Example</h2>

<pre>
String message = "hello world";
System.out.println(message.toString()); // Noncompliant;
</pre>

<h2>Compliant Solution</h2>

<pre>
String message = "hello world";
System.out.println(message);
</pre>
