<p>
All classes extend <code>Object</code> implicitly. Doing so explicitly is redundant, and redundant declarations needlessly clutter the code and can be confusing.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
public class Foo extends Object { // Noncompliant
  //...
}
</pre>

<h2>Compliant Solution</h2>

<pre>
public class Foo {
  //...
}
</pre>
