<p>Creating a new <code>Random</code> object each time a random value is needed is inefficient and produces only mediocre randomness. For better
efficiency and randomness, create a single <code>Random</code>, then store, and reuse it.</p>
<p>For even better, less-guessable random values, create, store and reuse a <code>SecureRandom</code> instead.</p>
<p>This rule finds cases where a new <code>Random</code> or <code>SecureRandom</code> is created each time a method is invoked.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public void doSomethingCommon() {
  Random rand = new Random();  // Noncompliant; new instance created with each invocation
  int rValue = rand.nextInt();
  //...
</pre>
<h2>Compliant Solution</h2>
<pre>
private Random rand = SecureRandom.getInstanceStrong();  // SecureRandom is preferred to Random

public void doSomethingCommon() {
  int rValue = this.rand.nextInt();
  //...
</pre>
<h2>Exceptions</h2>
<p>A class which uses a <code>Random</code> in its constructor and nowhere else will be ignored by this rule.</p>
<h2>See</h2>
<ul>
  <li> OWASP Top 10 2017 Category A6 - Security Misconfiguration </li>
</ul>

