<p>This rule allows banning certain methods.</p>

<p>Given parameters:</p>
    <ul>
      <li>className:java.lang.String</li>
      <li>methodName: replace</li>
      <li>argumentTypes: java.lang.CharSequence, java.lang.CharSequence</li>
    </ul>
<h2>Noncompliant Code Example</h2>
<pre>
String name;
name.replace("A","a");  // Noncompliant
</pre>