<p>Having a variable with the same name in two unrelated classes is fine, but do the same thing within a class hierarchy and you'll get confusion at best, chaos at worst.</p>

<p>This rule ignores <code>private</code> parent class members, but in all other such cases, the child class field should be renamed.</p>

<h2>Noncompliant Code Example</h2>

<pre>
public class Fruit {
  private Season ripe;

  // ...
}

public class Raspberry extends Fruit {
  private boolean ripe;  // Noncompliant
}
</pre>

<h2>Compliant Solution</h2>

<pre>
public class Fruit {
  private Season ripe;

  // ...
}

public class Raspberry extends Fruit {
  private boolean ripened;
}
</pre>
