<p>
  Many consider <code>clone</code> and <code>Cloneable</code> broken in Java,
  largely because the rules for overriding <code>clone</code> are tricky and difficult to get right.
  A copy constructor or copy factory should be used instead.
</p>
<p>
  This rule raises an issue when <code>clone</code> is overridden,
  whether or not <code>Cloneable</code> is implemented.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
public class MyClass {
  // ...

  public Object clone() { // Noncompliant
    //...
  }
}
</pre>

<h2>Compliant Solution</h2>
<pre>
public class MyClass {
  // ...

  MyClass (MyClass source) {
    //...
  }
}
</pre>
