<p>The Java 8 version of <code>HashMap</code> handles key clashes by storing nodes in a binary tree when more than 11 keys clash with each other, and
that tree needs to know the relative order of the keys. If you don't provide a <code>compareTo</code> method, <code>System.identityHashCode()</code>
will be used as the fallback, and that typically returns a value based on the object's memory location. The result could be a serious performance
degradation.</p>
<p><strong>Note</strong> that this rule is automatically disabled when the project's <code>sonar.java.source</code> is lower than <code>8</code>.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class Key {
  @Override
  public boolean equals(Object obj) {
    /* ... */
  }

  @Override
  public int hashCode() {
    /* ... */
  }
}

public void doTheThing() {
  Map &lt;Key,String&gt; map = new HashMap&lt;&gt;();  // Noncompliant
  // ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class Key implements Comparable{
  @Override
  public boolean equals(Object obj) {
    /* ... */
  }

  @Override
  public int hashCode() {
    /* ... */
  }

  @Override
  public int compareTo(Object o) {
    //...
  }
}

public void doTheThing() {
  Map &lt;Key,String&gt; map = new HashMap&lt;&gt;();
  // ...
}
</pre>

