<p>Shared naming conventions allow teams to collaborate efficiently. This rule raises an issue when a test class name does not match the provided regular expression.</p>

<h2>Noncompliant Code Example</h2>
<p>With the default value: <code>^[A-Z][a-zA-Z0-9]*Test$</code></p>
<pre>
class Foo {  // Noncompliant
}
</pre>

<h2>Compliant Solution</h2>
<pre>
class FooTest {
}
</pre>

