<p>The use of any value but <code>"validate"</code> for <code>hibernate.hbm2ddl.auto</code> may cause the database schema used by your application to
be changed, dropped, or cleaned of all data. In short, the use of this property is risky, and should only be used in production with the
<code>"validate"</code> option, if it is used at all.</p>
<h2>Noncompliant Code Example</h2>
<h2>Compliant Solution</h2>

