<p>When two methods have the same implementation, either it was a mistake - something else was intended - or the duplication was intentional, but may
be confusing to maintainers. In the latter case, one implementation should invoke the other.</p>
<h2>Noncompliant Code Example</h2>
<pre>
private final static String CODE = "bounteous";

public String getCode() {
  return CODE;
}

public String getName() {  // Noncompliant
  return CODE;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
private final static String CODE = "bounteous";

public String getCode() {
  return CODE;
}

public String getName() {
  return getCode();
}
</pre>
<h2>Exceptions</h2>
<p>Following methods are ignored: </p>
<ul>
  <li> Empty methods (with only <code>return;</code>, or without instructions); </li>
  <li> Methods only returning constants (<code>true</code>, <code>false</code>, <code>this</code>, <code>null</code>, <code>-1</code>, <code>0</code>,
  <code>1</code>, <code>""</code>) ; </li>
  <li> Methods only throwing an <code>Exception</code>; </li>
  <li> Methods only calling another method without parameter (for example: <code>throwMyCustomException()</code>). </li>
</ul>

