<p>All classes extending <code>org.apache.struts.action.Action</code> are potentially remotely reachable. The <code>ActionForm</code> object provided
as a parameter of the <code>execute</code> method is automatically instantiated and populated with the HTTP parameters. One should review the use of
these parameters to be sure they are used safely. </p>
<p>This rule is there to allow a security auditor to quickly find some potential hotspots to review.</p>
<h1>Noncompliant Code Example</h1>
<pre>
// Struts 1.1+
public final class CashTransferAction extends Action {

  public String fromAccount = "";
  public String toAccount = "";

  public ActionForward execute(ActionMapping mapping, ActionForm form, HttpServletRequest req, HttpServletResponse res) throws Exception {
    // usage of the "form" object to call some services doing JDBC actions
    [...]
    return mapping.findForward(resultat);
  }
}
</pre>
<h2>See</h2>
<ul>
  <li> OWASP Top 10 2017 Category A1 - Injection </li>
</ul>

