<p>This rule verifies that single-line comments are not located at the end of a line of code. The main idea behind this rule is that in order to be really readable, trailing comments would have to be properly written and formatted (correct alignment, no interference with the visual structure of the code, not too long to be visible) but most often, automatic code formatters would not handle this correctly: the code would end up less readable. Comments are far better placed on the previous empty line of code, where they will always be visible and properly formatted.</p>
<h2>Noncompliant Code Example</h2>

<pre>
int a1 = b + c; // This is a trailing comment that can be very very long
</pre>
<h2>Compliant Solution</h2>

<pre>
// This very long comment is better placed before the line of code
int a2 = b + c;
</pre>
<h2>Exceptions</h2>

<p>By default, the property "legalTrailingCommentPattern" allows to ignore comments containing only one word :</p>
<pre>
doSomething(); //FIXME
</pre>

